python/lsst/pipe/tasks/htmIndexEngine.py

```diff
@@ -133,6 +133,135 @@ def run(self, parsedCmd):
+        return {'pixel_id':pixel_id}
+
+    @staticmethod
+    def compute_coord(row, ra_name, dec_name):
+        return afwCoord.IcrsCoord(row[ra_name]*afwGeom.degrees,
+                                  row[dec_name]*afwGeom.degrees)
+
+    def _set_flags(self, record, row):
+        names = record.schema.getNames()
+        for flag in self._flags:
+            if flag in names:
+                attr_name = 'is_{}_name'.format(flag)
+                record.set(flag, bool(row[getattr(self.config, attr_name)]))
+
+    def _set_mags(self, record, row):
+        for item in self.config.mag_column_list:
+            record.set(item+'_flux', fluxFromABMag(row[item]))
+        if len(self.config.mag_err_column_map) > 0:
+            for err_key in self.config.mag_err_column_map.keys():
+                error_col_name = self.config.mag_err_column_map[err_key]
+                record.set(err_key+'_fluxSigma', fluxErrFromABMagErr(row[error_col_name], row[err_key]))
+
+    def _set_extra(self, record, row):
+        for extra_col in self.config.extra_col_names:
+            record.set(extra_col, row[extra_col])
+
+    def _fill_record(self, record, row, rec_num):
+        record.setCoord(self.compute_coord(row, self.config.ra_name, self.config.dec_name))
+        if self.config.id_name:
+            record.setId(row[self.config.id_name])
+        else:
+            rec_num += 1
+            record.setId(rec_num)
+        # No parents
+        record.setParent(-1)
+
+        self._set_flags(record, row)
+        self._set_mags(record, row)
+        self._set_extra(record, row)
+        return rec_num
+
+    def get_catalog(self, dataId, schema):
+        if self.butler.datasetExists(self.config.ref_dataset_name, dataId=dataId):
+            return self.butler.get(self.config.ref_dataset_name, dataId=dataId)
+        return afwTable.SourceCatalog(schema)
+
+    def make_schema(self, dtype):
+        mag_column_list = self.config.mag_column_list
+        mag_err_column_map = self.config.mag_err_column_map
+        if len(mag_err_column_map) > 0 and (not len(mag_column_list) == len(mag_err_column_map)
+                or not sorted(mag_column_list) == sorted(mag_err_column_map.keys())):
+            raise ValueError("Every magnitude column must have a corresponding error column")
+        # makes a schema with a coord, id and parent_id
+        schema = afwTable.SourceTable.makeMinimalSchema()
+
+        def add_field(name):
+            if dtype[name].kind == 'S':
+                # dealing with a string like thing.  Need to get type and size.
+                at_type = afwTable.aliases[str]
+                at_size = dtype[name].itemsize
+                return schema.addField(name, type=at_type, size=at_size)
+            else:
+                at_type = afwTable.aliases[dtype[name].type]
+                return schema.addField(name, at_type)
+
+        for item in mag_column_list:
+            key = schema.addField(item+'_flux', float)
```

Comment:
`key` is unused.  Why are you dropping it?  You need it if this is not going to be dreadfully slow.